rudderstack_blueprints/verify_sync_status.py

```diff
@@ -0,0 +1,81 @@
+import sys
+import argparse
+import requests
+import shipyard_utils as shipyard
+
+
+EXIT_CODE_SYNC_STATUS_SUCCESS = 0
+EXIT_CODE_BAD_REQUEST = 201
+EXIT_CODE_SYNC_STATUS_ERROR = 210
+EXIT_CODE_SYNC_STATUS_INCOMPLETE = 211
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--access-token', dest='access_token', required=True)
+    parser.add_argument('--source-id', dest='source_id', required=False)
+    args = parser.parse_args()
+    return args
+
+
+def get_source_status(source_id, access_token):
+    api_headers = {
+        'authorization': f"Bearer {access_token}",
+        'Content-Type': 'application/json'
+    }
+    api_url = "https://api.rudderstack.com/v2/sources"
+    source_status_url = api_url + f"/{source_id}/status"
+    source_status_json = {}
+    status_code = EXIT_CODE_SYNC_STATUS_SUCCESS
+    
+    # get response from API
+    try:
+        source_status_response = requests.get(source_status_url, 
+                                           headers=api_headers)
+        # check if successful, if not return error message
+        if source_status_response.status_code == requests.codes.ok:
+            source_status_json = source_status_response.json()
+        else:
+            print(f"Source status check failed. Reason: {source_status_response.content}")
+            sys.exit(EXIT_CODE_BAD_REQUEST)
+    except Exception as e:
+        print(f"Source {source_id} status check failed due to: {e}")
+        sys.exit(EXIT_CODE_BAD_REQUEST)
+    
+    # check source sync status
+    if source_status_json['status'] == 'finished':
+        print("Successfully managed to check sync")
+        if source_status_json.get('error'):
+            print(f"sync for sourceId: {source_id} failed with error: {source_status_json['error']}")
+            sys.exit(EXIT_CODE_SYNC_STATUS_ERROR)
+    else:
+        print(f"Sync for {source_id} is incomplete. Status {source_status_json['status']}")
+        status_code = EXIT_CODE_SYNC_STATUS_INCOMPLETE
```

Comment:
Plus, every status should be assessed for based on the returned response data and the timestamps. Refer to [Fivetran Blueprints](https://github.com/shipyardapp/fivetran-blueprints/blob/main/fivetran_blueprints/check_sync_status.py)

We can't just take the status at face value. We have to verify that the startedAt time is after the trigger time. Otherwise, it's possible that the job wouldn't start and the status would be related to the last run.

Reply:
Separated the check and the handle status functions.

Unlike Fivetran though: Rudderstack verify Sync status has no times attached to it, just the job_id for that specific sync run, any outstanding task data and then the status itself which is 'finished' or some variation of incomplete. 
if for some reason the Sync run has an error, it will still indicate that it's finished the sync run but then will output the specific error for that run

Reply:
Example of sync status response:

`{'jobId': '2Awp922RqIbu2DAR4kGLrIQxZg5', 'tasks': [], 'status': 'finished'}`

The handling of sync run data for this particular function came directly from the rudderstack airflow source code (https://github.com/rudderlabs/rudder-airflow-provider/blob/main/rudder_airflow_provider/hooks/rudderstack.py) because  even i was initially confused with the data being returned and how to handle it.